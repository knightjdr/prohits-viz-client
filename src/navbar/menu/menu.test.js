import React from 'react';
import { render } from '@testing-library/react';

import Menu from './menu';

jest.mock('./menu-style');

const toggleMenu = jest.fn();

const renderElement = (props) => render(<Menu {...props} />);

describe('Navbar menu', () => {
  describe('closed', () => {
    const props = {
      links: ['analysis', 'help'],
      open: false,
      route: 'home',
      toggleMenu,
    };

    it('should match snapshot', () => {
      const { container } = renderElement(props);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should toggle menu on button click', () => {
      const { getByLabelText } = renderElement(props);
      toggleMenu.mockClear();
      getByLabelText('navigation menu').click();
      expect(toggleMenu).toHaveBeenCalled();
    });
  });

  describe('open', () => {
    const props = {
      links: ['analysis', 'help'],
      open: true,
      route: 'home',
      toggleMenu,
    };

    it('should match snapshot', () => {
      const { container } = renderElement(props);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render analysis link', () => {
      const { getByText } = renderElement(props);
      expect(getByText(/analysis/i)).toBeInTheDocument();
    });

    it('should render help link', () => {
      const { getByText } = renderElement(props);
      expect(getByText(/help/i)).toBeInTheDocument();
    });
  });
});
