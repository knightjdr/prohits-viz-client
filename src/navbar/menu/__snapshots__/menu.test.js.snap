// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar menu closed should match snapshot 1`] = `
<div>
  <div>
    <button
      aria-label="navigation menu"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bars fa-w-14 fa-lg fa-rotate-null fa-pull-null nav__menu"
        data-icon="bars"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          fill="currentColor"
        />
      </svg>
    </button>
    <ul
      style="transform: scale(0);"
    >
      <li>
        <a
          href="/analysis"
        >
          analysis
        </a>
      </li>
      <li>
        <a
          href="/help"
        >
          help
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Navbar menu open should match snapshot 1`] = `
<div>
  <div>
    <button
      aria-label="navigation menu"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bars fa-w-14 fa-lg fa-rotate-null fa-pull-null nav__menu"
        data-icon="bars"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          fill="currentColor"
        />
      </svg>
    </button>
    <ul
      style="transform: scale(1);"
    >
      <li>
        <a
          href="/analysis"
        >
          analysis
        </a>
      </li>
      <li>
        <a
          href="/help"
        >
          help
        </a>
      </li>
    </ul>
  </div>
</div>
`;
